test/ask/runtime/channel_broker_test.exs

```diff
@@ -241,6 +241,50 @@ defmodule Ask.Runtime.ChannelBrokerTest do
   end
 
   describe "Nuntium" do
+    defp activate_respondent(state, "sms", respondent, not_before, not_after) do
+      {_, state, _} =
+        ChannelBroker.handle_cast(
+          {:ask, "sms", respondent, "token", "foo", not_before, not_after},
+          state
+        )
+
+      state
+    end
+
+    test "Respect schedules for SMS" do
+      %{state: state, respondents: respondents, test_channel: test_channel} = build_survey("sms")
+
+      now = SystemTime.time().now
+      not_before = DateTime.add(now, 5, :second)
+      not_after = DateTime.add(now, 60, :second)
+      expired = DateTime.add(now, -5, :second)
+
+      with_mock Ask.Runtime.Survey, [contact_attempt_expired: fn _ -> :ok end] do
+        state =
+          state
+          |> activate_respondent("sms", Enum.at(respondents, 0), now, not_after)
+          |> activate_respondent("sms", Enum.at(respondents, 1), not_before, expired)
+          |> activate_respondent("sms", Enum.at(respondents, 2), not_before, expired)
+          |> activate_respondent("sms", Enum.at(respondents, 3), not_before, not_after)
+          |> activate_respondent("sms", Enum.at(respondents, 4), not_before, not_after)
+
+        expected_active_respondents = [
+          Enum.at(respondents, 0),
+          Enum.at(respondents, 3),
+          Enum.at(respondents, 4)
+        ]
+
+        # All respondents should be in broker queue, but only the first one contacted
```

Comment:
Where's Surveda guaranteeing this? We [activate respondents with a `not_before` in the following minute, too](https://github.com/instedd/surveda/blob/cb53f55d7bd973a5aa8f46c44a2b2a6e084df90f/lib/ask/runtime/channel_broker_state.ex#L149), so I think there's a good chance of contacting both respondents 3 and 4 here when the broker runs.

Reply:
It's an outdated comment from a previous iteration on this test, it's actually not true even in the context of this test. Will fix